semantic-core/src/Data/Core.hs

```diff
@@ -215,27 +216,27 @@ a .= b = send (a := b)
 infix 3 .=
 
 
-data Ann f a
-  = Ann Loc (f a)
+data Ann ann f a
+  = Ann ann (f a)
   deriving (Eq, Foldable, Functor, Generic1, Ord, Show, Traversable)
 
-instance HFunctor Ann
+instance HFunctor (Ann ann)
 
-instance RightModule Ann where
+instance RightModule (Ann ann) where
   Ann l b >>=* f = Ann l (b >>= f)
 
 
-ann :: (Carrier sig m, Member Ann sig) => HasCallStack => m a -> m a
+ann :: (Carrier sig m, Member (Ann Span) sig) => HasCallStack => m a -> m a
 ann = annWith callStack
 
-annAt :: (Carrier sig m, Member Ann sig) => Loc -> m a -> m a
-annAt loc = send . Ann loc
+annAt :: (Carrier sig m, Member (Ann ann) sig) => ann -> m a -> m a
+annAt ann = send . Ann ann
 
-annWith :: (Carrier sig m, Member Ann sig) => CallStack -> m a -> m a
-annWith callStack = maybe id annAt (stackLoc callStack)
+annWith :: (Carrier sig m, Member (Ann Span) sig) => CallStack -> m a -> m a
+annWith callStack = maybe id (annAt . snd) (stackLoc callStack)
 
 
-stripAnnotations :: (HFunctor sig, forall g . Functor g => Functor (sig g)) => Term (Ann :+: sig) a -> Term sig a
+stripAnnotations :: forall ann a sig . (HFunctor sig, forall g . Functor g => Functor (sig g)) => Term (Ann ann :+: sig) a -> Term sig a
```

Comment:
I’ve rearranged the type parameters to make it easier to reference the annotation type, now that `Ann` has been generalized over the annotation type.